scripts/schema-publish.sh

```diff
@@ -6,15 +6,69 @@
 
 schemaDir="src/schemas/validation"
 branch=$(git branch --show-current)
-version=${branch:1:3}
-echo === Building schemas into ./deploy/oas/$version
+
+if [[ $branch =~ ^v([0-9]+\.[0-9]+\.[0-9]+) ]]; then
+  deploydir="./deploy/oas/${BASH_REMATCH[1]}"
+else
+  deploydir="./deploy-preview"
+fi
+
+# create the date-stamped schemas
+publish_schema() {
+  local schema="$1"
+  local date="$2"
+  local sedCmd="$3"
+
+  local base=$(basename $schema '.yaml')
+  local target=$deploydir/$base/$date
+
+  mkdir -p $deploydir/$base
+  # replace the WORK-IN-PROGRESS placeholders
+  sed -e $sedCmd $schemaDir/$schema > $target.yaml
+
+  node scripts/yaml2json/yaml2json.js "$target.yaml"
+  rm "$target.yaml"
+  mv "$target.json" "$target"
+
+  # Find the jekyll lander markdown file for this iteration.
+  
+  # Define jekyllLander by referencing a markdown file whose name matches the
+  # iteration date. This approach would work with today's jekyll landers
+  # local jekyllLander=$deploydir/$base/$date.md
+  
+  # Define jekyllLander by referencing a static file
+  # local jekyllLander=$deploydir/$base/latest.md
+  
+  # Define jekyllLander by finding the first markdown file already-present in the
+  # deploy directory.
+  # local jekyllLander=$(find "$deploydir/$base" -maxdepth 1 -name "*.md" | head -n 1)
+
+  # Define jekyllLander by finding all the markdown files already-present in the
+  # deploy directory. If multiple files are present, this script will fail as it
+  # attempts to move these multiple files to a single $target.md
+  # This behavior is the closest match to that of the previous version of the
+  # schema-publish script.
```

Comment:
```suggestion
```
Let's not confuse the next maintainers with possible future implementation alternatives.

Let's instead discuss the future implementation in #4152.

Reply:
Absolutely. This was really intended as placeholder for discussion. Seems well-enough captured here now to be referenced in #4152. I'll pull all this cruft.